var/spack/repos/builtin/packages/hpcg/package.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+import platform
+
+
+class Hpcg(AutotoolsPackage):
+    """HPCG is a software package that performs a fixed number of multigrid
+    preconditioned (using a symmetric Gauss-Seidel smoother) conjugate gradient
+    (PCG) iterations using double precision (64 bit) floating point values."""
+
+    homepage = "https://www.hpcg-benchmark.org"
+    url = "http://www.hpcg-benchmark.org/downloads/hpcg-3.1.tar.gz"
+    git = "https://github.com/hpcg-benchmark/hpcg.git"
+
+    version('develop', branch='master')
+    version('3.1', sha256='33a434e716b79e59e745f77ff72639c32623e7f928eeb7977655ffcaade0f4a4')
+
+    variant('openmp', default=True, description='Enable OpenMP support')
+
+    patch('missing_variable.patch')
```

Comment:
```suggestion
    patch('https://github.com/hpcg-benchmark/hpcg/commit/e9e0b7e6cae23e1f30dd983c2ce2d3bd34d56f75.patch', sha256='23b9de83042eb7a8207fdddcfa79ae2cc1a17e8e623e2224c7751d7c328ee482', when='%gcc@9:')
```
This way we do not have to add the full patch to the repository.

According to the comment at https://github.com/hpcg-benchmark/hpcg/commit/e9e0b7e6cae23e1f30dd983c2ce2d3bd34d56f75, this doesn't work with older GCCs, so we should probably add a `when` clause.